tests/test_urls.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+# Download and build the data if it does not exist.
+
+import unittest
+import importlib
+import os
+import warnings
+import sys
+
+TO_SKIP = [
+    './parlai/tasks/__pycache__',
+    './parlai/tasks/interactive',
+    './parlai/tasks/fromfile',
+    './parlai/tasks/taskntalk',
+    './parlai/tasks/integration_tests',
+    './parlai/tasks/dialog_babi_plus',
+    './parlai/tasks/decanlp',
+]
+
+SPECIFIC_BUILDS = {
+    './parlai/tasks/opensubtitles': ['build_2009', 'build_2018'],
+    './parlai/tasks/coco_caption': ['build_2014', 'build_2015', 'build_2017'],
+}
+
+
+class TestUtils(unittest.TestCase):
+    def test_http_response(self):
+        sys.path.insert(0, './lib')
+        tasks = [f.path for f in os.scandir('./parlai/tasks') if f.is_dir()]
+        for task in tasks:
+            if task in TO_SKIP:
+                continue
+            with warnings.catch_warnings():
```

Comment:
What's the point of this? what warnings are being hit?

Reply:
The warnings are:

`/anaconda3/lib/python3.7/site-packages/socks.py:58: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
  from collections import Callable`

`
/anaconda3/lib/python3.7/socket.py:660: ResourceWarning: unclosed <socket.socket fd=9, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('192.168.1.14', 61030), raddr=('171.64.67.140', 443)>
  self._sock = None
`

I have tried to address the ResourceWarning, but after research I have found out that it is a trivial error and most people bypass it by simply muting the error. (I do explicitly close the requests session in `check_header`)

Reply:
OKay, thanks for clarifying.